### ISSUE TITLE
[bilibili] Cannot download videos with 1080p resolution

### ISSUE THREAD

**Issue by TAKUBUNTU** [yt-dlp/yt-dlp#5874]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

all

### Provide a description that is worded well enough to be understood

Videos with resolution only up to 720p can be downloaded in 720p, but videos with resolution of 1080p or higher cannot be downloaded, not even those below 720p.
In other issues, it is assumed that the difference is whether the account is Premier or not, but even regular members who are not Premier can watch 1080p. Only downloading is not possible.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
>yt-dlp -vU https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[debug] Command-line config: ['-vU', 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32']
[debug] Encodings: locale cp932, fs utf-8, pref cp932, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109451-geb0455d646-20221225 (setts), ffprobe N-109451-geb0455d646-20221225
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P 高清, 360P 流畅, 480P 清晰, 720P 高清 are missing; you have to login or become premium member to download them
[BiliBili] 13255517: Extracting chapters
ERROR: [BiliBili] BV1Dx411n7Pg_p32: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1485, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1582, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1641, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2573, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 1035, in raise_no_formats
yt_dlp.utils.ExtractorError: [BiliBili] BV1Dx411n7Pg_p32: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by TAKUBUNTU** [comment]
This issue seems to happen with playlists.
For stand-alone videos, I was able to download videos with a resolution of 1080p.

**Comment by dirkf** [comment]
There is a format in the retrieved JSON that isn't being extracted, presumably the FLV that used to be the only extracted format:
```diff
--- old/yt_dlp/extractor/bilibili.py
+++ new/yt_dlp/extractor/bilibili.py
@@ -18,6 +18,7 @@
     make_archive_id,
     mimetype2ext,
     parse_count,
+    parse_resolution,
     parse_qs,
     qualities,
     srt_subtitles_timecode,
@@ -49,18 +50,19 @@
         } for audio in audios]
 
         formats.extend({
-            'url': traverse_obj(video, 'baseUrl', 'base_url', 'url'),
+            'url': traverse_obj(video, 'baseUrl', 'base_url', 'url', 'backup_url'),
             'ext': mimetype2ext(traverse_obj(video, 'mimeType', 'mime_type')),
             'fps': float_or_none(traverse_obj(video, 'frameRate', 'frame_rate')),
             'width': int_or_none(video.get('width')),
-            'height': int_or_none(video.get('height')),
+            'height': int_or_none(video.get('height'))
+                      or traverse_obj(parse_resolution(format_names.get(video.get('id') or play_info.get('quality'))), 'height'),
             'vcodec': video.get('codecs'),
             'acodec': 'none' if audios else None,
             'tbr': float_or_none(video.get('bandwidth'), scale=1000),
             'filesize': int_or_none(video.get('size')),
-            'quality': int_or_none(video.get('id')),
-            'format': format_names.get(video.get('id')),
-        } for video in traverse_obj(play_info, ('dash', 'video', ...)))
+            'quality': int_or_none(video.get('id') or play_info.get('quality')),
+            'format_id': format_names.get(video.get('id')) or play_info.get('format'),
+        } for video in traverse_obj(play_info, ('dash', 'video', ...), 'durl'))
 
         missing_formats = format_names.keys() - set(traverse_obj(formats, (..., 'quality')))
         if missing_formats:
@@ -297,7 +299,7 @@
         is_anthology = len(page_list_json) > 1
 
         part_id = int_or_none(parse_qs(url).get('p', [None])[-1])
-        if is_anthology and not part_id and self._yes_playlist(video_id, video_id):
+        if is_anthology and part_id is None and self._yes_playlist(video_id, video_id):
             return self.playlist_from_matches(
                 page_list_json, video_id, title, ie=BiliBiliIE,
                 getter=lambda entry: f'https://www.bilibili.com/video/{video_id}?p={entry["page"]}')
```
Then:
```shellsession
$ python3.9 -m yt_dlp -v -F 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32'
[debug] Command-line config: ['-v', '-F', 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b644025b] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: c73355510
[debug] Python 3.9.15 (CPython i686 32bit) - Linux-4.4.0-210-generic-i686-with-glibc2.23 (OpenSSL 1.1.1s  1 Nov 2022, glibc 2.23)
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3
[debug] Optional libraries: Cryptodome-3.11.0, certifi-2019.11.28, secretstorage-3.2.0, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1735 extractors
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P 高清, 360P 流畅, 720P 高清 are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
[BiliBili] 13255517: Extracting chapters
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for BV1Dx411n7Pg_p32:
ID     EXT RESOLUTION │   FILESIZE PROTO │ VCODEC  ACODEC
──────────────────────────────────────────────────────────
flv480 flv 480p       │  158.90MiB https │ unknown unknown
$
```

**Comment by TAKUBUNTU** [comment]
I'm sorry.
My point is why can I watch 1080p videos but not download 1080p videos?
I want to download 1080p videos.
I left out the cookies.txt file above, but I also retrieved and wrote the cookies.txt.

**Comment by dirkf** [comment]
Possibly if someone who was a member tried the problem URL using browser cookies it would become apparent why the 1080p format isn't being found. Currently formats are being pulled from the `.dash.video` branch of the JSON retrieved from the page, but perhaps there is some further location that should be considered. This is what the JSON suggests might be available, of which we are only finding the `32` quality:
```py
 'support_formats': [{'codecs': ['hev1.1.6.L120.90'],
                      'display_desc': '1080P',
                      'format': 'flv',
                      'new_description': '1080P 高清',
                      'quality': 80,
                      'superscript': ''},
                     {'codecs': ['avc1.640028', 'hev1.1.6.L120.90'],
                      'display_desc': '720P',
                      'format': 'flv720',
                      'new_description': '720P 高清',
                      'quality': 64,
                      'superscript': ''},
                     {'codecs': ['avc1.64001F', 'hev1.1.6.L120.90'],
                      'display_desc': '480P',
                      'format': 'flv480',
                      'new_description': '480P 清晰',
                      'quality': 32,
                      'superscript': ''},
                     {'codecs': ['avc1.64001E', 'hev1.1.6.L120.90'],
                      'display_desc': '360P',
                      'format': 'mp4',
                      'new_description': '360P 流畅',
                      'quality': 16,
                      'superscript': ''}],
```

**Comment by Vangelis66** [comment]
`'quality': 80`, which is the one corresponding to the `1080p` vertical resolution (OP is after), is being available **ONLY as HEVC video stream inside the FLV media container**, a standards-incompatible combination; see related PR #5823 for specifics... 

However, this doesn't explain why **720p/FLV** and/or **360p/MP4** aren't being detected (because for these, compatible `videocodec+container` combinations **do** exist) ...

**Comment by Vangelis66** [comment]
FWIW, **without having an account with bilibili**, upstream `youtube-dl` is still able to fetch the `avc720p30/FLV` variant (but only that): 
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-vF', 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32']
[debug] Encodings: locale cp1253, fs mbcs, out cp737, pref cp1253
[debug] youtube-dl version 2022.12.27.19419
[debug] Python version 3.4.4 (CPython) - Windows-Vista-6.0.6003-SP2
[debug] exe versions: ffmpeg 5.0, ffprobe 5.0, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] 1Dx411n7Pg: Downloading video info page
[info] Available formats for 1Dx411n7Pg:
format code  extension  resolution note
0            flv        unknown    272.99MiB
```
and 
```
youtube-dl "https://www.bilibili.com/video/BV1Dx411n7Pg?p=32" => 

[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] 1Dx411n7Pg: Downloading video info page
[download] Destination: 【Gaki使】Gaki no Tsukai 2009（936-986）-1Dx411n7Pg.flv
[download] 100% of 272.99MiB in 06:37
```
MediaInfo log: 
```
Format                                   : Flash Video
File size                                : 273 MiB
Duration                                 : 24 min 9 s
Overall bit rate                         : 1 580 kb/s
Tagging application                      : Version 1.9
description                              : Bilibili VXCode Swarm Transcoder v0.3.42_sync_fix

Video
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L4
Format settings                          : CABAC / 6 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 6 frames
Codec ID                                 : 7
Duration                                 : 24 min 9 s
Bit rate                                 : 1 455 kb/s
Width                                    : 1 280 pixels
Height                                   : 720 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 30.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.053
Stream size                              : 258 MiB (94%)
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
```

**Comment by dirkf** [comment]
The yt-dl extractor uses http://interface.bilibili.com/v2/playurl but it's quite confused about the various formats available. The yt-dlp extractor looks in the JSON sent in the page itself.

**Comment by TAKUBUNTU** [comment]
Thank you all so much for your varied opinions.
I am totally unskilled and am able to download the video thanks to you all.
Thank you so much.

**Comment by TAKUBUNTU** [comment]
https://www.bilibili.com/video/BV1Ax411H7oY?p=47&vd_source=0d7810556d0cfe3c3c4072ae70d73822

This video could be downloaded in 1080p.

**Comment by TAKUBUNTU** [comment]
>>[Vangelis66](https://github.com/Vangelis66)
youtube-dl "https://www.bilibili.com/video/BV1Dx411n7Pg?p=32" => 

[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] 1Dx411n7Pg: Downloading video info page
[download] Destination: 【Gaki使】Gaki no Tsukai 2009（936-986）-1Dx411n7Pg.flv
[download] 100% of 272.99MiB in 06:37



I tried this and it actually downloaded
"https://www.bilibili.com/video/BV1Dx411n7Pg?p=1".
It seems that the top of the playlist is downloaded.

**Comment by Vangelis66** [comment]
> This video could be downloaded in **1080p**.

... Presumably via providing cookies from a logged-in (premium?) session 😉 ... 
In any case, `yt-dlp` **still** has a semi-broken `bilibiliIE` 😞 : 

The video in OP is **NOT downloadable at all**: 
```
yt-dlp -F "https://www.bilibili.com/video/BV1Dx411n7Pg?p=32" => 

[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P ??, 360P ??, 480P ??, 720P ?? are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp for how to manually pass cookies
[BiliBili] 13255517: Extracting chapters
ERROR: [BiliBili] BV1Dx411n7Pg_p32: No video formats found!;
```
whereas in my browser I can watch `480p` & `360p` formats **without being signed-in**: 

![BB1](https://user-images.githubusercontent.com/9669492/209890732-712ea7b4-fe84-4151-b6c4-4d5d56a3f1e2.png)

(`1080p` & `720p` available to account holders).

The second video cited behaves as expected with regards to `yt-dlp`: 
```
yt-dlp -F "https://www.bilibili.com/video/BV1Ax411H7oY?p=47" => 

[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Ax411H7oY?p=47
[BiliBili] 1Ax411H7oY: Downloading webpage
[BiliBili] BV1Ax411H7oY: Extracting videos in anthology
[BiliBili] Format(s) 1080P ??, 720P ?? are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp for how to manually pass cookies
[BiliBili] 13300737: Extracting chapters
[info] Available formats for BV1Ax411H7oY_p47:
ID EXT RESOLUTION FPS |   FILESIZE  TBR PROTO | VCODEC         VBR ACODEC      ABR
----------------------------------------------------------------------------------
0  m4a audio only     | ~ 11.90MiB  67k https | audio only         mp4a.40.2   67k
1  m4a audio only     | ~ 22.49MiB 127k https | audio only         mp4a.40.2  127k
2  m4a audio only     | ~ 22.49MiB 127k https | audio only         mp4a.40.2  127k
3  mp4 480x360     30 | ~ 62.50MiB 353k https | avc1.64001E   353k video only
4  mp4 480x360     30 | ~ 50.90MiB 288k https | hev1.1.6.L120 288k video only
5  mp4 640x480     30 | ~139.90MiB 790k https | avc1.64001E   790k video only
6  mp4 640x480     30 | ~114.63MiB 648k https | hev1.1.6.L120 648k video only
```
with `1080p` & `720p` downloadable when signed-in ...

**Comment by Vangelis66** [comment]
> I tried this and it actually downloaded
> "https://www.bilibili.com/video/BV1Dx411n7Pg?p=1".
> It seems that the top of the playlist is downloaded.

... You are correct 😉 , I didn't realise at the time that the `youtube-dl`-downloaded video, 

`【Gaki使】Gaki no Tsukai 2009（936-986）-1Dx411n7Pg.flv`

actually has media content identical to 

`【Gaki使】Gaki no Tsukai 2009（936-986） p01 Gaki no Tsukai #0936 (2009.01.04) - 山崎VSモリマン 未公開 [BV1Dx411n7Pg_p1].mp4`

fetched with `yt-dlp` via: 
```
yt-dlp -f 6 "https://www.bilibili.com/video/BV1Dx411n7Pg/?p=1"
```
I'll probably have to delete my previous comment above, so as **not to create false expectations** 😞 ; my apologies...
As one grows older, memory lapses occur... Thanks to you, I now remembered a [previous comment of mine](https://github.com/ytdl-org/youtube-dl/issues/31217#issuecomment-1234969710) submitted in the upstream issue tracker; IOW, what you found out is **upstream** issue [#31051](https://github.com/ytdl-org/youtube-dl/issues/31051) - and its (more recent) duplicate [#31217](https://github.com/ytdl-org/youtube-dl/issues/31217) :smile: ...

**Comment by pukkandan** [comment]
> In any case, `yt-dlp` **still** has a semi-broken `bilibiliIE` 😞 :
> 
> The video in OP is **NOT downloadable at all**:


Even with the patch in https://github.com/yt-dlp/yt-dlp/issues/5874#issuecomment-1365452076 ?

**Comment by dirkf** [comment]
Interesting. I merged the two extraction tactics in my yt-dl back-port and got this:
```shellsession
$ python -m youtube_dl -v  'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32' -F[debug] System config: [u'--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32', u'-F']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 774b346f9
[debug] Python version 2.7.18 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] 21733763: Downloading video info page
[BiliBili] 21733763: Downloading video info page
[BiliBili] 21733763: Downloading video info page
[BiliBili] 13255517: Extracting chapters
[info] Available formats for BV1Dx411n7Pg_p32:
format code  extension  resolution note
mp4          mp4        unknown    72.75MiB
flv480       flv        480p       158.90MiB
flv720       flv        unknown    304.17MiB
flv          flv        unknown    60.37MiB (best)
$
```
The `best` format gives this _mpv_ output,
```
 (+) Video --vid=1 (h264 1280x720 30.000fps)
 (+) Audio --aid=1 (aac 2ch 48000Hz)
File tags:
 Description: Packed by Bilibili XCoder v1.0(fixed_gap:True)
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1280x720 yuv420p
AV: 00:00:47 / 00:03:43 (21%) A-V:  0.000 Dropped: 5 Cache: 85s/27MB
```
whereas the 720p format gives this:
```
 (+) Video --vid=1 (h264 1280x720 30.000fps)
 (+) Audio --aid=1 (aac 2ch 48000Hz)
File tags:
 Description: Bilibili VXCode Swarm Transcoder v0.2.49(gap_fixed:False)
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1280x720 yuv420p
AV: 00:00:07 / 00:23:55 (0%) A-V:  0.000 Cache: 109s/28MB
```
So the `best` format, which should be 1080p, is actually 720p and missing the last 20:12 of the video. In fact the "1080p" video has 5 parts @ 720p. Initially, such multi-part formats could be skipped.

**Comment by nihil-admirari** [comment]
[FFmpeg-Builds](https://github.com/yt-dlp/FFmpeg-Builds) just got [HEVC-over-FLV support](https://github.com/yt-dlp/FFmpeg-Builds/commit/c1ba17b6ceefa8326d831fb896b03150e61fc74b). It would be interesting to test it on real-world examples, so if someone has premium access to BiliBili, your help would be greatly appreciated.

**Comment by Vangelis66** [comment]
> Even with the patch in https://github.com/yt-dlp/yt-dlp/issues/5874#issuecomment-1365452076 ?

Apologies, I meant `yt-dlp` in its **current** state (log below generated by running from git-source): 
```
[debug] Command-line config: ['-vF', 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.12.30 [e107c2b8c]
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.7.16 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL 1.1.1s  1 Nov 2022)
[debug] exe versions: none
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1752 extractors
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P ??, 360P ??, 480P ??, 720P ?? are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
[BiliBili] 13255517: Extracting chapters
ERROR: [BiliBili] BV1Dx411n7Pg_p32: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
(Traceback omitted for brevity)
```
After applying the referenced patch by @dirkf :
```
[debug] Command-line config: ['-vF', 'https://www.bilibili.com/video/BV1Dx411n7Pg?p=32']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.12.30 [e107c2b8c]
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.7.16 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL 1.1.1s  1 Nov 2022)
[debug] exe versions: none
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1752 extractors
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P ??, 360P ??, 720P ?? are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
[BiliBili] 13255517: Extracting chapters
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for BV1Dx411n7Pg_p32:
ID     EXT RESOLUTION |   FILESIZE PROTO | VCODEC  ACODEC
----------------------------------------------------------
flv480 flv 480p       |  158.90MiB https | unknown unknown
```
and... 
```[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Dx411n7Pg?p=32
[BiliBili] 1Dx411n7Pg: Downloading webpage
[BiliBili] BV1Dx411n7Pg: Extracting videos in anthology
[BiliBili] Format(s) 1080P ??, 360P ??, 720P ?? are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
[BiliBili] 13255517: Extracting chapters
[info] BV1Dx411n7Pg_p32: Downloading 1 format(s): flv480
[download] Destination: 【Gaki使】Gaki no Tsukai 2009（936-986） p32 Gaki no Tsukai #0967 (2009.08.09) - 温厚タレント大集合！ 芸能界怒り王グランプリ!! 中盤 [BV1Dx411n7Pg_p32].flv
[download] 100% of  158.90MiB in 00:03:44 at 724.33KiB/s
```
Post download, it's found that: 

**VCODEC** = AVC High@L3.1
**ACODEC** = AAC LC

**Comment by lockmatrix** [comment]
This is a flv format video, you need this PR #6438 .
And you need to login and `--cookies-from-browser chrome` to download 1080p

**Comment by ralioc** [comment]
I am noticing that anything I try to download with yt-dlp on youtube over the last week stopped downloading resolutions above 720p. Been using the same config file for the last couple of years. I first noticed the 4k content that was available on certain channels was only grabbing 720p on/around the 15th of March. I'm trying to go back and see when it actually started happening so that's an estimate. The videos are viewable in 1080 and 4k on the webpage and through other downloaders. It's only yt-dlp that is having this recent problem. Any ideas?

**Comment by Vangelis66** [comment]
> anything I try to download with `yt-dlp` on **youtube**

...Then why have you posted in a [**bilibili**] issue/thread?

**Comment by z0z0r4** [comment]
这么说 FLV 1080P 的问题已经被解决了？

**Comment by bashonly** [comment]
> So the FLV 1080P video can be downloaded now? Should I update yt-dlp?

No, a PR will need to be finished and merged before yt-dlp is able to download it

**Comment by z0z0r4** [comment]
> > So the FLV 1080P video can be downloaded now? Should I update yt-dlp?
> 
> No, a PR will need to be finished and merged before yt-dlp is able to download it

Anyway, I don't know much about this project, if any PR is waiting to be merged?

Or still no one?

I need to download these old videos... In high quality, so if this is indeed the case I will try to open one... 

Although I don't know the operational architecture of yt-dlp, etc...

**Comment by z0z0r4** [comment]
> > > > So the FLV 1080P video can be downloaded now? Should I update yt-dlp?
> > > 
> > > 
> > > No, a PR will need to be finished and merged before yt-dlp is able to download it
> > 
> > 
> > Anyway, I don't know much about this project, if any PR is waiting to be merged?
> > Or still no one?
> > I need to download these old videos... In high quality, so if this is indeed the case I will try to open one...
> > Although I don't know the operational architecture of yt-dlp, etc...
> > 我不了解这个项目…只是使用着。
> > 请问有贡献者正在解决这个问题吗？如果没有，因为我急需下载它们，我会试着创建个PR来解决，尽管我从未试过参与yt-dlp的开发。
> 
> 试试https://greasyfork.org/ja/scripts/413228-bilibili%E8%A7%86%E9%A2%91%E4%B8%8B%E8%BD%BD

 比起这个油猴找到有更好的下载器，只是希望ytdlp支持...

**Comment by grqz** [comment]
#9117 has been merged, should this be closed now?

**Comment by bashonly** [comment]
Closed by 1d6ab17d0752ee9cf19e3e63c7dec7b600d3f228
